proposals/introduce_ebpf.md

```diff
@@ -0,0 +1,82 @@
+# Introduce ebpf
+
+- **Authors**: Eileen Yu (@Eileen-Yu), Jim Bugwadia (jim@nirmata.com)
+- **Created**: Feb 15th, 2023
+- **Updated**:
+- **Abstract**: Introduce ebpf to trigger kyverno policy for extended use cases
+
+## Contents
+
+- [Introduction](#introduction)
+- [The Problem](#the-problem)
+- [Proposed Solution](#proposed-solution)
+- [Open Question](#open-question)
+
+## Introduction
+
+Kyverno is now running as a dynamic admission controller for policy management. This allows users to manage Kubernetes native resources.
+
+This proposal aims to extend the ability of kyverno by introduing ebpf, so that it could cover events at a finer granularity at runtime.
+
+## The Problem
+
+At current stage kyverno focus on enforcing policies based on admission webhook. This enable users to define policy using k8s manifests to easily manage resources.
+
+Well, it is common for users to raise further fine-grained requirements out of security concern. This goes far beyond simple "approve / deny" the request.
+One instance is, users may want to limit the volume mount to a specific container. Another example is to control the access to certain API paths based on layer 7 calls.
```

Comment:
>users may want to limit the volume mount to a specific container

But this is totally possible today with a Kyverno validate rule, or is there some nuance here which isn't said?

Reply:
>Another example is to control the access to certain API paths based on layer 7 calls

Can you explain what you mean here? "layer 7 calls" doesn't make sense to me in this context.

Reply:
> But this is totally possible today with a Kyverno validate rule, or is there some nuance here which isn't said?

The main difference is we wish to block the `violation` at runtime, rather than check the static manifest to decide whether to apply it. So even if there are resources that `bypassed` the rule, they can still be caught and shut down.

Reply:
> Can you explain what you mean here? "layer 7 calls" doesn't make sense to me in this context.

Thx for pointing out! 

ebpf can be aware of network calls in the application level, compared with the network policy which mainly around TCP/IP. 

Updated in the docs.